pkg/cmd/cli/cmd/login/login.go

```diff
@@ -192,6 +193,30 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 
 // RunLogin contains all the necessary functionality for the OpenShift cli login command
 func RunLogin(cmd *cobra.Command, options *LoginOptions) error {
+	contextName := kcmdutil.GetFlagString(cmd, "context")
+	if len(contextName) > 0 {
+		if _, exists := options.StartingKubeConfig.Contexts[contextName]; exists {
+			options.StartingKubeConfig.CurrentContext = contextName
+			if err := kclientcmd.ModifyConfig(options.PathOptions, *options.StartingKubeConfig, true); err != nil {
+				return err
+			}
+			return nil
+		}
```

Comment:
If the context provided is not valid, then the command either attempts to use the value from `--cluster`, if one was give, or resumes normal execution. Not sure if that is the right approach, or if an error should be returned here instead.